core/sql/sqlcomp/CmpSeabaseDDLrepos.cpp

```diff
@@ -516,6 +585,62 @@ short CmpSeabaseDDL::upgradeReposComplete(ExeCliInterface * cliInterface,
 
         case 1:
           {
+            // If there were views on the old Repository tables, they are
+            // still there, by virtue of the fact that we did "SKIP VIEW CHECK"
+            // on the ALTER TABLE RENAME. Now we will capture their view
+            // definitions (which will contain the old table name, not the
+            // renamed table name) and replay that against the new Repository
+            // table. If the replay fails, we treat that as an unrecoverable
+            // situation and ignore it. Instead, we'll save the view definition
+            // text in the TEXT table with a different TEXT_TYPE and clue the
+            // user in that it is there. They can then try to create the view
+            // at their leisure. Note that they may have to change the view
+            // definition themselves, e.g. if we dropped a column from the
+            // repository table that their view happened to reference.
+            //
+            // Note that this work is done in one step because the only state
+            // we can depend upon across redrives to this method is the substep
+            // number. Any in-memory list would be lost across redrives.
+            NABoolean someViewSaved = FALSE;
```

Comment:
I didn't unit test this case (and I think I should), but I think the code works correctly in this case. What will happen is that when we process the first repository table referenced, the view will get dropped and recreated. When we get to the second repository table, the view will no longer exist since we already dropped it.